### PR TITLE
Mon 34004 associate otl metrics to host service

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description
REFS:MON-34004

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: engine/inc/com/centreon/engine/configuration/applier/command.hh [hunk 1]
```diff
@@ -34,22 +33,25 @@ class state;
 
 namespace applier {
 class command {
+  void _create_command(const configuration::command& conf_obj);
+
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/inc/com/centreon/engine/configuration/applier/command.hh:37]: done

File: engine/inc/com/centreon/engine/configuration/applier/command.hh [hunk 2]
```diff
@@ -34,22 +33,25 @@ class state;
...
  public:
   command();
+
+  command(command const& right) = delete;
+  command& operator=(command const& right) = delete;
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/inc/com/centreon/engine/configuration/applier/command.hh:42]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/host_serv_extractor.hh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+};
+
+template <typename host_set, typename service_set>
+host_serv_metric host_serv_extractor::is_allowed(
+    const host_set& hosts,
+    const service_set& services) const {
+  return _host_serv_list->is_allowed(hosts, services);
+}
+
+class host_serv_attributes_extractor : public host_serv_extractor {
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/host_serv_extractor.hh:70]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/host_serv_extractor.hh [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** For more information : contact@centreon.com
+*/
+
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/host_serv_extractor.hh:18]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/data_point_fifo.hh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ */
+#ifndef CCE_MOD_OTL_SERVER_DATA_POINT_FIFO_HH
+#define CCE_MOD_OTL_SERVER_DATA_POINT_FIFO_HH
+
+#include "otl_data_point.hh"
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/data_point_fifo.hh:21]: to do in the simplified PR

File: engine/inc/com/centreon/engine/commands/command.hh [hunk 6]
```diff
@@ -109,6 +114,29 @@ class command {
                    uint32_t timeout,
                    result& res) = 0;
 
+  /**
+   * @brief connector and host serv extractor share a list of host serv witch is
```
Comment (bouda1) [inline engine/inc/com/centreon/engine/commands/command.hh:118]: witch=sorcière
which is better in your case :-D

**Review by bouda1 — APPROVED** [review]

File: engine/inc/com/centreon/engine/commands/command.hh (hunk 6 above)
Comment (jean-christophe81) [reply engine/inc/com/centreon/engine/commands/command.hh]: done

**Review by bouda1 — APPROVED** [review]

**Pull request merged**